graphblas/vector.py

```diff
@@ -708,6 +708,49 @@ def apply(self, op, right=None, *, left=None):
             size=self._size,
         )
 
+    def select(self, op, thunk=None):
+        """
+        GrB_Vector_select
+        Compute SelectOp at each element of the calling Vector, keeping
+        elements which return True.
+        """
+        method_name = "select"
+        if thunk is None:
+            thunk = False  # most basic form of 0 when unifying dtypes
+        if type(thunk) is not Scalar:
+            dtype = self.dtype if self.dtype._is_udt else None
+            try:
+                thunk = Scalar.from_value(thunk, dtype, is_cscalar=None, name="")
+            except TypeError:
+                thunk = self._expect_type(
+                    thunk,
+                    Scalar,
+                    within=method_name,
+                    keyword_name="thunk",
+                    extra_message="Literal scalars also accepted.",
+                    op=op,
+                )
+        op = get_typed_op(op, self.dtype, thunk.dtype, is_right_scalar=True, kind="select")
```

Comment:
We should only include `self.dtype` here when the operator is not positional (similar as prev comment).